src/hotspot/share/oops/resolvedMethodEntry.cpp

```diff
@@ -23,9 +23,25 @@
  */
 
 #include "cds/archiveBuilder.hpp"
+#include "cppstdlib/type_traits.hpp"
 #include "oops/method.hpp"
 #include "oops/resolvedMethodEntry.hpp"
 
+STATIC_ASSERT(std::is_trivially_copyable_v<ResolvedMethodEntry> == true);
+#ifdef _LP64
+# ifdef ASSERT
+STATIC_ASSERT(sizeof(ResolvedMethodEntry) == 32);
+# else
+STATIC_ASSERT(sizeof(ResolvedMethodEntry) == 24);
+# endif
+#else
+# ifdef ASSERT
+STATIC_ASSERT(sizeof(ResolvedMethodEntry) == 20);
+# else
+STATIC_ASSERT(sizeof(ResolvedMethodEntry) == 16);
+# endif
+#endif
```

Comment:
I think this can be cleaned up as:

```
#ifdef _LP64
STATIC_ASSERT(sizeof(ResolvedMethodEntry) == DEBUG_ONLY(32) NOT_DEBUG(24));
#else
STATIC_ASSERT(sizeof(ResolvedMethodEntry) == DEBUG_ONLY(20) NOT_DEBUG(16));
#endif
```

But I think this will be better without the need to hard code numbers:

```
// There should be no more padding at the end of ResolvedMethodEntry
class ResolvedMethodEntryWithExtra : public ResolvedMethodEntry {
  u1 _extra_field;
};
STATIC_ASSERT(sizeof(ResolvedMethodEntryWithExtra) > sizeof(ResolvedMethodEntry));
```

I tested by changing `u4 _padding2` to `u4 _padding2` in `ResolvedMethodEntry` and the static assert fails.